Fix localize theme feature links

d-alleyne opened the pull request:
## Proposed Changes

* The links in the Features section aren't localized on loggeg-out home pages. This commit fixes this.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Visit `/es/theme/nook` while logged out, and verify that the `Features`/`Funcionalidades` links are localized.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

d-alleyne committed:
Ensure Features section of Theme page has localized links

d-alleyne committed:
Bugfix

Pass the logged in state

yuliyan commented on client/my-sites/theme/theme-features-card.js:
```diff
@@ -1,13 +1,18 @@
 
 const ThemeFeaturesCard = ( { isWpcomTheme, siteSlug, features, translate, onClick } ) => {
+	const localizeUrl = useLocalizeUrl();
+	const { isLoggedIn } = useIsLoggedIn();
```
I might have mislead you with the link I sent you in our discussion - I think it will be more appropriate to use the [isUserLoggedIn](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/current-user/selectors.js#L15) selector instead.

yuliyan commented on client/my-sites/theme/theme-features-card.js:
```diff
@@ -27,7 +32,15 @@ const ThemeFeaturesCard = ( { isWpcomTheme, siteSlug, features, translate, onCli
+								<a
+									href={ localizeUrl(
+										`https://wordpress.com/themes/filter/${ term }/${ siteSlug || '' }`,
+										getLocaleSlug(),
```
JFYI, if you pass [a falsy value for the locale parameter, it should pick it up from the provider](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-utils/src/localize-url.tsx#L269-L272). It's fine to pass it explicitly, but it can save you some extra imports.

yuliyan commented on client/my-sites/theme/theme-features-card.js:
```diff
@@ -27,7 +32,15 @@ const ThemeFeaturesCard = ( { isWpcomTheme, siteSlug, features, translate, onCli
-								<a href={ `/themes/filter/${ term }/${ siteSlug || '' }` }>{ name }</a>
+								<a
+									href={ localizeUrl(
+										`https://wordpress.com/themes/filter/${ term }/${ siteSlug || '' }`,
```
With the switch to full URL, I'm unable to test the client-side navigation as it's now an external link for the dev and calypso.live environments (due to running from a different host).

I came across the following helper for the themes paths - https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/themes/helpers.js#L61. It might be worth trying if it can be used in this case.

yuliyan reviewed (changes requested):
I've tested the changes on both local dev environment and calypso.live, but the locale slug doesn't appear to be prepended to the URL path as I expect it to be:

![a1h4H8kMmATTaYLF](https://github.com/Automattic/wp-calypso/assets/2722412/205ee040-96de-4e6c-9b86-b5f2e49b0c00)

d-alleyne commented:
> I've tested the changes on both local dev environment and calypso.live, but the locale slug doesn't appear to be prepended to the URL path as I expect it to be:
> 
> 

I'm experiencing similar behaviour, where:
```jsx
<a href={ `https://wordpress.com/es/themes/filter/${ term }/${ siteSlug || '' }` }>
```
shows the wordpress URL when logged in, but the relative path when logged out

d-alleyne committed:
Implement code review suggestion

yuliyan commented on client/my-sites/theme/theme-features-card.js:
```diff
@@ -18,19 +22,22 @@ const ThemeFeaturesCard = ( { isWpcomTheme, siteSlug, features, translate, onCli
-						</li>
-					) ) }
+					{ features.map( ( { name, slug, term } ) => {
+						const themePath = localizeThemesPath(
```
Nit: The variable name `themePath` seems a little inaccurate given that it refers to the themes filter path.

yuliyan reviewed (commented):
Same as before, I'm not seeing the locale in the filters path name:

![x4bTYtMq9Cx4NGKX](https://github.com/Automattic/wp-calypso/assets/2722412/8440adf7-b5b8-4ce7-b1e3-4784f2ce62ba)

I can see the locale slug in the path name only after navigating to a different theme after the initial load, so the problem is probably only on the SSR.

d-alleyne committed:
Retrieve the locale from the component props

Using `getLocaleSlug` always returned `en` on the server, while
returning the correct value on the browser.

This approach uses the fact the component is localized, and is always
given the locale as a component property prior to rendering.

yuliyan reviewed (approved):
Works as described, LGTM!
